### PR TITLE
[BUG] fix bug 4076: `PerformanceWarning` in `load_from_tsfile_to_dataframe`

### LINKED ISSUE #4076: [BUG] PerformanceWarning while loading multivariate dataset.
**Describe the bug**
Warning while loading Duckduckgeese dataset.

```
PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead.  To get a de-fragmented frame, use `newframe = frame.copy()`
  data["dim_" + str(dim)] = instance_list[dim]
```

**To Reproduce**

```python
from sktime.datasets import load_UCR_UEA_dataset
train_x, train_y = load_UCR_UEA_dataset("DuckDuckGeese")
```

**Expected behavior**

Shouldn't be any warning.

**Additional context**

**Versions**
<details>

Screenshot 2023-01-06 at 15 16 04

</details>

### PR DESCRIPTION
**Pull request by ericjb**
Fixes #4076. 

I had a lengthy discussion with my coding buddy (Copilot). [lengthy = Copilot's first 5 suggestions failed; 6th seems to produce identical output to the original code, without the PerformanceWarning]

I did extensive testing [ = I ran this on the contributed example. "extensive testing" is a trademark :-)]

### CODE REVIEW

**Review by fkiraly — DISMISSED** [review]
Yes, I can confirm by inspection that the output should be the same.

The warning is probably caused by assigning the type in advance and assigning columns in a loop, instead of constructing in a single go.

We should probably add a `dtype="float64"` in the `pd.Series` constructors hough?

**Comment by fkiraly** [comment]
(this code smells badly - there are up to 10 indentations in it!)

also, you ought to fix code formatting. `black .`

**Comment by ericjb** [comment]
I forgot to do the pre-commit before the PR. Doing it now. Will resubmit.

File: sktime/datasets/_readers_writers/ts.py [hunk 1]
```diff
@@ -599,9 +599,9 @@ class values.
         elif metadata_started and data_started and len(instance_list) == 0:
             raise OSError("file contained metadata but no data")
         # Create a DataFrame from the data parsed above
-        data = pd.DataFrame(dtype=np.float32)
-        for dim in range(0, num_dimensions):
-            data["dim_" + str(dim)] = instance_list[dim]
+        data_dict = {f'dim_{dim}': pd.Series(instance_list[dim]) for dim in range(num_dimensions)}
```
Comment (VascoSch92) [inline sktime/datasets/_readers_writers/ts.py:602]: You could put a comment with the reference to the issue ;)

**Review by fkiraly — DISMISSED** [review]
Thanks!

Added `float` typing and removed accidental changes to a notebook. Hope that is fine?

**Review by fkiraly — APPROVED** [review]

**Comment by ericjb** [comment]
Hi @fkiraly - regarding "Removed accidental changes to a notebook. Hope that is fine?" 
My local pre-commit was failing on this notebook which was totally unrelated to my PR. (In a separate module.) I suppose the right thing to do was to either (1) configure my local pre-commit to ignore this file or (2) if the notebook is passing the sktime linting then configure my local pre-commit to match the rules of the sktime linters. I did not know how to do either of these so I just cleaned up the offending notebook. Nothing "accidental" about it - although perhaps not advisable.
Please advise on how to achieve either (1) or (2) so I don't repeat problems due to my local pre-commit being different than the sktime linting rules. Thanks in advance!

**Comment by fkiraly** [comment]
Just saw this - apologies, messages in closed PR do not notify me

not sure how to solve this, might be useful to diagnose in a live session (eg, tech/meetup)
basically, you want the linter to lint only on change

**Pull request merged**